### PR TITLE
feat(DateSelect): Add Date Select component DEV-53

### PR DESCRIPTION
**Pull request by giubatt**
- Added `DateSelect`
- Refactored `Selector` component into `Select`
  - `Select` uses `Select.Option` as children to define the options instead of an `options` prop
- Deprecated `Selector` component

### CODE REVIEW

File: src/components/DateSelect/DateSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+
+const DateSelect = ({
+  id,
+  label,
+  placeholder,
+  value,
+  defaultValue,
+  disabled = false,
+  onChange,
+  hightlightFilled = true,
```
[the remark this answers is not in the document]
Comment (giubatt) [reply src/components/DateSelect/DateSelect.tsx]: Fixed in 5db96f3

File: src/components/DateSelect/DateSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+              >
+                <label
+                  htmlFor="dateFrom"
+                  sx={{
+                    variant: "text.body1",
+                    color: "text.40",
+                    textAlign: "end",
+                  }}
+                >
+                  From
```
Comment (VagnerNico) [inline src/components/DateSelect/DateSelect.tsx:174]: We need to discuss the i18n approach to those library components texts.

File: src/components/DateSelect/DateSelect.tsx [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+      defaultValue={defaultValue as SelectProps["defaultValue"]}
+      disabled={disabled}
+      onChange={(value) => {
+        if (value) onChange?.((value as unknown) as DateValue)
+      }}
+      highlightFilled={highlightFilled}
+      fullWidth={fullWidth}
+    >
+      {children}
+      <SelectOptionCustom
```
Comment (VagnerNico) [inline src/components/DateSelect/DateSelect.tsx:122]: This custom option can be optional, default rendered but removed with some flag.

File: src/components/Select/Select.tsx [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+import { Button, Popover, Icon } from ".."
+import { InputGroupContext } from "../InputGroup/Context"
+import { getDataFromChildren, isArrayEqual } from "./utils"
+import { SelectOption } from "./SelectOption"
+import { SelectContext } from "./context"
+
+export type SelectValue = string | number | Record<string, unknown>
+export type SelectedItems = Array<SelectValue>
+
+export type SelectOption = {
```
Comment (VagnerNico) [inline src/components/Select/Select.tsx:16]: You have two variables with the same name SelectOption.

**Review by VagnerNico — CHANGES_REQUESTED** [review]

File: src/components/Select/Select.tsx (hunk 4 above)
Comment (giubatt) [reply src/components/Select/Select.tsx]: Fixed in 1029c24

File: src/components/DateSelect/DateSelect.tsx (hunk 3 above)
Comment (giubatt) [reply src/components/DateSelect/DateSelect.tsx]: Added option in cd14c96

**Review by VagnerNico — APPROVED** [review]
Approved

**Review by welitonderesende — APPROVED** [review]

**Pull request merged**